### ISSUE TITLE
Last page in profile using moshidon leads to infinite pagination

### ISSUE THREAD

**Issue by maybeanerd** [dahlia/hollo#48]
When using moshidon, viewing profiles (my own, but also from other servers) when scrolling to the end, pagination continues to fetch the last page forever.
Cannot reproduce this on Mastodon when using the same version of moshidon, so it seems related to hollos responses.

https://github.com/user-attachments/assets/1cec5023-08ec-459f-b0e9-ba561733f53b

**Comment by ThisIsMissEm** [comment]
@dahlia I think this is because pagination is using `gte` and `lte` on the posts, not `gt` and `lt`?

```
-        query.max_id == null ? undefined : lte(posts.id, query.max_id),
-        query.min_id == null ? undefined : gte(posts.id, query.min_id),
+        query.max_id == null ? undefined : lt(posts.id, query.max_id),
+        query.min_id == null ? undefined : gt(posts.id, query.min_id),
```

**Comment by ThisIsMissEm** [comment]
For reference this is the mastodon implementation: 
- https://github.com/mastodon/mastodon/blob/ebfeaebedbc43526116fc5c0abe82f3d8745c927/app/models/account_conversation.rb#L60
- https://github.com/mastodon/mastodon/blob/ab7816a4141e88cf7e05ba49638ee95fcc6f71ff/app/models/concerns/paginable.rb#L7

Logic:

- If min_id is supplied, then:
  - gt min_id
  - lt max_id if supplied
  - reverse
- else
  - lt max_id if supplied
  - gt since_id if supplied

**Comment by dahlia** [comment]
I guess @ThisIsMissEm is right. I'm fixing it soon.

**Comment by dahlia** [comment]
Fixed in 0.1.2.  Update to [`ghcr.io/dahlia/hollo:0.1.2`](https://github.com/dahlia/hollo/pkgs/container/hollo/294285018?tag=0.1.2) or just [`latest`](https://github.com/dahlia/hollo/pkgs/container/hollo/294285018?tag=latest).

**Comment by maybeanerd** [comment]
Can confirm, fixed the issue for me. Thank you both for being so quick ❤️
